namespace-provisioner/use-case7.hbs.md

```diff
@@ -0,0 +1,469 @@
+
+   - `host`, `username`, `caFile` and `password` or `personal access token` values for HTTP-based Git Authentication.
+   - `ssh-privatekey`, `identity`, `identity_pub`, and `known_hosts` for SSH-based Git Authentication.
+
+    >**Note:** The `stringData` key of the secret must end with either **.yaml** or **.yml** suffix.
+
+    Using HTTP(s) based Authentication
+    : If using Username and Password for authentication.
+
+
+       Assuming that, in this configuration for an air-gapped environment,
+       the Git Repository server has a custom certificate of authority that
+       cannot be verified against public issuers, you must provide
+       the `caFile` content to log in against it.
+
+
+      ```yaml
+      cat << EOF | kubectl apply -f -
+      apiVersion: v1
+      kind: Secret
+      metadata:
+        name: workload-git-auth
+        namespace: tap-install
+      type: Opaque
+      stringData:
+        content.yaml: |
+          git:
+            #! For HTTP Auth. Recommend using https:// for the git server.
+            host: GIT-SERVER
+            username: GIT-USERNAME
+            password: GIT-PASSWORD
+            caFile: |
+              -----BEGIN CERTIFICATE-----
+              ...
+              -----END CERTIFICATE-----
+      EOF
+      ```
+
+    Using SSH based Authentication
+    : If you prefer using an SSH private key for authentication, create the Git secret with the authentication details as follows:
+
+
+      ```yaml
+      cat << EOF | kubectl apply -f -
+      apiVersion: v1
+      kind: Secret
+      metadata:
+        name: workload-git-auth
+        namespace: tap-install
+      type: Opaque
+      stringData:
+        content.yaml: |
+          git:
+            host: GIT-SERVER
+            #! For SSH Auth
+            ssh_privatekey: SSH-PRIVATE-KEY
+            identity: SSH-PRIVATE-KEY
+            identity_pub: SSH-PUBLIC-KEY
+            known_hosts: GIT-SERVER-PUBLIC-KEYS
+      EOF
+      ```
+
+2. To create a secret to be added to the service account in the developer namespace within the GitOps repository, you can use this [example](https://github.com/vmware-tanzu/application-accelerator-samples/blob/main/ns-provisioner-samples/gitops-airgap/resources/git.yaml) for HTTP-based or this [example](https://github.com/vmware-tanzu/application-accelerator-samples/blob/main/ns-provisioner-samples/gitops-airgap/resources/settings-xml.yaml) for `setings.xml`-based, or follow the provided example below.
+
+    Rather than directly including the actual username and password in the Git repository secret,
+    utilize the `data.values.imported` keys to add references to the values from the `git-auth` secret
+    created in Step 1.
+
+    This secret represents the actual Git secret that will be created by the Namespace Provisioner
+    in each managed namespace. It should be included in your Git repository linked in the
+    `additional_sources` section of `tap-values.yaml` mentioned in Step 4.
```

Comment:
Should this be '.. mentioned in step 3.' or '.. mentioned in the next step'?

Reply:
You are right, it should refer to next step